contrib/config_option_database/config_database.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python3
+#############################################################################
+# Copyright (c) 2019 Balabit
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 2 as published
+# by the Free Software Foundation, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# As an additional exemption you are allowed to compile & link against the
+# OpenSSL libraries as published by the OpenSSL project. See the file
+# COPYING for details.
+#
+#############################################################################
+
+from pathlib import Path
+import json
+
+from utils.ConfigOptions import get_driver_options
+from argparse import ArgumentParser
+
+
+def parse_args():
+    parser = ArgumentParser()
+    parser.add_argument('context', type=str, help='source/destination')
+    parser.add_argument('driver', type=str, help='driver')
+    parser.add_argument('--rebuild', '-r', action='store_true')
+    args = parser.parse_args()
+    return (args.context, args.driver, args.rebuild)
+
+
+def build_db():
+    db = {'source': {}, 'destination': {}}
+    for context, driver, keyword, arguments, parents in get_driver_options():
+        db[context].setdefault(driver, {'options': [], 'blocks': {}})
+        add_to = db[context][driver]
```

Comment:
It is purposely general. It changes with the parents of the option, so it does not stay a `driver_node` if the option has parents.

Reply:
With this context, could you please suggest another name? I couldn't think of one :)

Reply:
What about `database_options_block` or `database_options_node` or `database_options_level`? 
It is not easy, because we are jumping between options normal level and options block levels.